great_expectations/datasource/data_connector/file_path_data_connector.py

```diff
@@ -160,6 +164,11 @@ def _get_batch_definition_list_from_batch_request(
             )
         )
 
+        if len(self.sorters) > 0:
+            batch_definition_list = self._sort_batch_definition_list(
+                batch_definition_list=batch_definition_list
```

Comment:
Moving sorting occur before we do anything with `index`.  In cases where the `data_connector_query` contains an index + sorter was configured to do highest --> lowest, we would run into situations where we were only sorting 1 value. 

*Note* This change is what results in the test changes below.

Reply:
Makes sense! Same question as above wrt passing which sorter to use into the method.